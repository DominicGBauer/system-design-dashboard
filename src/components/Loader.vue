<template>
  <div v-if="isLoading" class="loader">
    <img src="../assets/loader.svg" />
  </div>
</template>

<script>
export default {
  name: 'Loadder',
  props: {
    isLoading: Boolean,
  },
}
</script>

<style lang="scss" scoped>
.loader {
  display: flex;
  justify-content: center;

  img {
    width: 120px;
  }
}
</style>
