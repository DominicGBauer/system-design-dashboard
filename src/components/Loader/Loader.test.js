import { mount } from '@vue/test-utils'
import Loader from '@/components/Loader/Loader.vue'

describe('Loader.vue', () => {
  it('renders the Loader', () => {
    const wrapper = mount(Loader, {
      props: {
        isLoading: true,
      },
    })
    expect(wrapper.classes()).toContain('loader')
  })

  it('renders the Loader and it looks the same', () => {
    const wrapper = mount(Loader, {})
    expect(wrapper.element).toMatchSnapshot()
  })
})
