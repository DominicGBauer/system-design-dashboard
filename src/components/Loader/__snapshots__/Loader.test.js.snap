// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader.vue renders the Loader and it looks the same 1`] = `<!--v-if-->`;
